lib/omnibus/licenses.rb

```diff
@@ -0,0 +1,206 @@
+#
+# Copyright 2015 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'uri'
+require 'fileutils'
+
+module Omnibus
+  class Licenses
+    include Logging
+
+    OUTPUT_DIRECTORY = "LICENSES".freeze
+
+    class << self
+      # @see (Licenses#create!)
+      def create!(project)
+        new(project).create!
+      end
+    end
+
+    #
+    # The project to create licenses for.
+    #
+    # @return [Project]
+    #
+    attr_reader :project
+
+    #
+    # @param [Project] project
+    #   the project to create licenses for.
+    #
+    def initialize(project)
+      @project = project
+    end
+
+    #
+    # Creates the license files for given project.
+    # It is assumed that the project has already been built.
+    #
+    def create!
+      prepare
+      create_software_license_files
+      create_project_license_file
+    end
+
+    #
+    # Creates the required directories for licenses.
+    #
+    def prepare
+      FileUtils.rm_rf(output_dir)
+      FileUtils.mkdir_p(output_dir)
+    end
+
+    #
+    # Creates the top level license file for the project.
+    # Top level file is created at #{project.license_file_path}
+    # and contains the name of the project, version of the project,
+    # text of the license of the project and a summary of the Licenses
+    # of the included software components.
+    #
+    def create_project_license_file
+      File.open(project.license_file_path, 'w') do |f|
+        f.puts "#{project.name} #{project.build_version} license: \"#{project.license}\""
+        f.puts ""
+        f.puts project_license_content
+        f.puts ""
+        f.puts components_license_summary
+      end
+    end
+
+    #
+    # Copies the license files specified by the software components into the
+    # output directory.
+    #
+    def create_software_license_files
+      license_map.each do |name, values|
+        license_files = values[:license_files]
+
+        license_files.each do |license_file|
+          if license_file && is_local(license_file)
+            input_file = File.expand_path(license_file, values[:project_dir])
+            output_file = license_package_location(name, license_file)
+            FileUtils.cp(input_file, output_file)
```

Comment:
This should not have an affect on existing projects. If a project does not have any licensing information we will create a `LICENSE` file under its install directory that say `"This project has an 'Unspecified' license."` But we can turn it off if that is undesirable behavior. 

The only case that will fail is when the specified `license_file` is not there and I will handle that case now :smile:.